File: core/src/main/java/com/a65apps/core/interactors/persons/PersonListRepository.java
package com.a65apps.core.interactors.persons;

import com.a65apps.core.entities.Person;

import java.util.List;

import io.reactivex.rxjava3.annotations.Nullable;
import io.reactivex.rxjava3.core.Single;

public interface PersonListRepository {
    Single<List<Person>> getAllPersons(@Nullable final String searchString);
}

File: core/src/main/java/com/a65apps/core/interactors/persons/PersonListRepository.kt
package com.a65apps.core.interactors.persons

import com.a65apps.core.entities.Person
import kotlinx.coroutines.flow.Flow

interface PersonListRepository {
    fun getAllPersonsFlow(searchString: String): Flow<List<Person>>
    suspend fun getAllPersons(searchString: String): List<Person>
}
